src/test/java/utils/ParserTest.java

```diff
@@ -0,0 +1,122 @@
+package utils;
+
+import commands.AddCommand;
+import commands.DeleteCommand;
+import commands.ListCommand;
+import commands.ExitCommand;
+import commands.Command;
+import exceptions.DukeException;
+import org.junit.jupiter.api.Test;
+
+import static org.junit.jupiter.api.Assertions.assertNotNull;
+import static org.junit.jupiter.api.Assertions.assertTrue;
+import static org.junit.jupiter.api.Assertions.assertThrows;
+import static org.junit.jupiter.api.Assertions.fail;
+
+
+class ParserTest {
```

Comment:
Colour me impressed. 6 JUnit tests instead of the minimum 3 shows dedication to refining and perfecting your code. The tests are also modular in nature for your 'ParserTest' class; testing each of the commands 'add todo', 'add event', 'delete' etc. and also testing for various exceptions. All in all, this is a very comprehensive test set.